### PR TITLE
[Feature] New benchmark test suite.

### PR DESCRIPTION
**Pull request by frettini**
# Description

Adding a new benchmark test suite and infrastructure to the project that will enable continuous tracking of eradiate's performance. This will help find regressions and have a better understanding of the impact of our changes to the performance of the project. 

The benchmark infrastructure is based on the ASV package which runs statistical benchmark tests. The package allows users to test the project over a range of commits and environments, and includes a browser GUI and regression tests. The package has been integrated in the `benchmarks` directory. 

The tests written are directly taken from the existing regression tests. The tests were refactored so that both the regression tests and the benchmark test both call the same library of test cases. 

Additionally, a new click CLI was added as a thin wrapper to the ASV wrapper. While ASV defaults to using a git-ref and cloning the project to benchmark it, our CLI defaults to using the current environment for tests. This favours a manual testing approach as we release new features on branches such as main or next. 
The CLI also include an update and custom version of ASVDb that lets us archive results to an external database. This will be useful to archive results to a central ASV database.

Note that the dependencies have changed and were updated accordingly. 

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `next` branch
- [ ] I updated the change log if relevant
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

File: src/eradiate/test_tools/test_cases/atmospheres.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import numpy as np
+
+from eradiate.experiments import AtmosphereExperiment
+from eradiate.units import unit_registry as ureg
```
Comment (leroyvn) [inline src/eradiate/test_tools/test_cases/atmospheres.py:4]: Prefer relative imports when you are in the `eradiate` package:
```suggestion
from ...experiments import AtmosphereExperiment
from ...units import unit_registry as ureg
```

File: src/eradiate/test_tools/test_cases/atmospheres.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                "thermoprops": {
+                    "identifier": "afgl_1986-us_standard",
+                    "z": np.linspace(0, 120, 61) * ureg.km,
+                    "additional_molecules": False,
+                },
+                "absorption_data": "monotropa",
+                "error_handler_config": absorption_database_error_handler_config,
+            },
+        },
+    )
```
Comment (leroyvn) [inline src/eradiate/test_tools/test_cases/atmospheres.py:140]: Add ending newline

**Review by leroyvn — CHANGES_REQUESTED** [review]
Hi @frettini, thanks for this PR. It's overall fine, I mostly comments on style and details.

File: benchmarks/benchmarks/__init__.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import sys
+import os
+
+if "ASV" in os.environ:
```
[the remark this answers is not in the document]
Comment (frettini) [reply benchmarks/benchmarks/__init__.py:4]: No, ASV sets those environment variables when it uses one of the environments it created i.e. doesn't use a specified existing environment.

File: benchmarks/eradiate.toml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+offline = false
+
+[absorption_database.error_handling]
+# This default configuration ignores bound errors on pressure and temperature
+# variables because this usually occurs at high altitude, where the absorption
+# coefficient is very low and can be safely forced to 0.
+p = {missing = "raise", scalar = "raise", bounds = "ignore"}
+t = {missing = "raise", scalar = "raise", bounds = "ignore"}
+# Ignore missing molecule coordinates, raise on bound error.
+x = {missing = "ignore", scalar = "ignore", bounds = "raise"}
```
[the remark this answers is not in the document]
Comment (frettini) [reply benchmarks/eradiate.toml:49]: I don't understand the suggested change, should I completely remove line 49?

File: requirements/layered.yml [hunk 5]
```diff
@@ -68,6 +68,7 @@ tests:
   packages:
     - "pytest"
     - "pytest-json-report"
+    - "asv>=0.6.*"
```
[the remark this answers is not in the document]
Comment (frettini) [reply requirements/layered.yml]: I have applied the change, does it need a rerun of the dependencies update process?

File: requirements/layered.yml [hunk 6]
```diff
@@ -68,6 +68,7 @@ tests:
   packages:
     - "pytest"
     - "pytest-json-report"
+    - "asv>0.6.*"
```
Comment (leroyvn) [inline requirements/layered.yml:71]: I made a mistake in my previous suggestion:
```suggestion
    - "asv=0.6.*"
```

File: src/eradiate/test_tools/benchmark/asvdb.py
Comment (leroyvn) [inline src/eradiate/test_tools/benchmark/asvdb.py:1]: As we discussed, adding a header here to mention where the code comes from would be good. Something like: "This is a vendored and updated copy of the asvdb project (https://github.com/rapidsai/asvdb)."

**Review by leroyvn — APPROVED** [review]
Hi @frettini, I think we're almost good to go. After you take care of the final details, we can merge!

File: requirements/layered.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (leroyvn) [reply requirements/layered.yml]: No yet, you've already updated the lock files and got something consistent.

**Comment by leroyvn** [comment]
I forced-pushed a couple of minor updates, after a squash. I removed the lock file update from the PR, I'll regenerate locks on `next` (after I also upgrade pip lock file management to the `uv` tool).

**Pull request merged**